httpclient-jetty/src/main/java/io/fabric8/kubernetes/client/jetty/JettyHttpClientBuilder.java

```diff
@@ -65,6 +68,11 @@ public JettyHttpClient build() {
       sharedWebSocketClient.setConnectTimeout(connectTimeout.toMillis());
     }
     sharedHttpClient.setFollowRedirects(followRedirects);
+    sharedHttpClient.setMaxConnectionsPerDestination(MAX_CONNECTIONS);
+    sharedWebSocketClient.getHttpClient().setMaxConnectionsPerDestination(MAX_CONNECTIONS);
```

Comment:
In looking at the websocket logic, the http connection pool appears only to be used for the establishment of the upgrade request - once the websocket is established it does not count against that pool.  I think your tests should pass without setting this, but of course setting it higher shouldn't be a problem either.

Reply:
Consider the wording for the test where the `simultaneous` word was chosen instead of `concurrent`. 
Despite being a very unlikely scenario, what I'm trying to ensure is that regardless of concurrency, or simultaneity, if a client opens `n` connections at the same time and later on they all upgrade simultaneously, the implementation should work.

```
----connect------upgrade---->
----connect------upgrade---->
vs.
-----connect-----upgrade---------->
----------------connect---upgrade->
-connect----upgrade--------------->
```

Reply:
So an additional test should be provided to assert this situation